### PR TITLE
CORE-19330: FlowTimeout Reason

### PR DESCRIPTION
**Pull request by jujoramos**
Add new 'reason' property to the 'FlowTimeout' schema so that the
platform can fully report to usser why a particular flow was timed out.

### CODE REVIEW

**Comment by jujoramos** [comment]
Runtime OS Change: https://github.com/corda/corda-runtime-os/pull/5496

File: gradle.properties [hunk 1]
```diff
@@ -5,7 +5,7 @@ cordaProductVersion = 5.2.0
 # NOTE: update this each time this module contains a breaking change
 ## NOTE: currently this is a top level revision, so all API versions will line up, but this could be moved to
 ##   a per module property in which case module versions can change independently.
-cordaApiRevision = 30
+cordaApiRevision = JR
```
Comment (vkolomeyko) [inline gradle.properties]:
```suggestion
cordaApiRevision = 31
```

**Review by vkolomeyko — DISMISSED** [review]
LGTM

File: gradle.properties (hunk 1 above)
Comment (jujoramos) [reply gradle.properties]: Yep, I'm waiting for reviews on both `PR`s before acquiring the lock and using the proper version. Thanks for pointing out anyways 👍

**Review by conal-smith — DISMISSED** [review]

**Review by conal-smith — APPROVED** [review]

**Review by vkolomeyko — APPROVED** [review]

**Pull request merged**